[Misc] Fix SPARC questionnaire csv import

acrowthe opened the pull request:
- Disabled pagination by default
- Fixed bug with conditional handling that caused operators `<>` and `<` to not be parsed
- Added arguments to questionnaire import script:
  - `--forms`: A list of the CSV files to import. Eg. `--forms "Your Voice Matters" PQ`
  - `--paginate`: If the questionnaire should be paginated. True or False, Default `False`
  - `--subject-types`: The subject types this questionnaire should be able to be created for. Default `/SubjectTypes/Patient/Visit`
  - `--logging`: The log level that should be run with. One of `run`, `error`, `warn` or `info`. Defaults to `info` if not provided or not recognized.
  - `--max-answers`: The default Max Answers for a question if not specified for that specific question. Default `1`
  - `--max-per-subject`: The maximum number of times these forms can be created for an individual subject. Default `1`
- Add `repeated section` meta section:
  - Requires the following columns:
    - `Question Type` = `repeated section`
    - `Section Name` = a section ID
    - `Reference Question` = the ID of a list question. This question must have the same parent as the repeated section
    - (optional) `Variable Name` = the displayed title of the generated sections
  - For each option in the referenced question a `conditional section` will be created
    - This `conditional section` will only show if the referenced question has that specific value
    - Each generated `conditional section` will contain a copy of every child of the `repeated section`

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
- Add additional logging for debugging nested section issues
- Disable default pagination
- Change list of imported forms to sparc forms
- Add parameters for options and list of forms to import

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
Add support for repeated sections.
- Repeated sections require:
  1. Question Type = "repeated section"
  2. Section Name = a section ID
  3. Reference Question = the ID of list question. This question must have the same parent
     as the repeated section
  4. (optional) Variable Name = the displayed title of the section

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
Fix conditional operators `<>` and `<` not working
Change default logging level from INFO to WARNING

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
- Fix conditional question processing leading to question properties not
  being assigned to the correct question
- Add log level argument

sdumitriu reviewed (approved):
Looks good to me, but I didn't test it.

acrowthe commented:
TODO:
- Change Questionnaire properties to be defined on command line rather than individually defined in code:
```
-v "property node name" "property value"
```
- Add way to add new generic columns to csv without needing to update the script
  - Potential implementation: Insert row 2 to .csv to specify which default handler should be used for this column

sashaandjic commented:
This conditional did not get imported:
![image](https://github.com/data-team-uhn/cards/assets/23243244/099f59a2-c175-4453-9545-45dd7b15b378)

from: https://docs.google.com/spreadsheets/d/1N2H6oLDf8fJkiZiM9u0xP7SURJuPiNHcNT7bOeY8aAU/edit#gid=1761023239
[Your Voice Matters.csv](https://github.com/data-team-uhn/cards/files/13422466/Your.Voice.Matters.csv)

sashaandjic commented:
Can we add support for `Enable Notes` = Ture/False?

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
- Fix ampersands '&' in strings breaking output XML
- Improve error handling if string could not be parsed to a number

marta- commented:
To do:
- support telephone and address answer types

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
- Fix issue with conditional parsing and capitalized conditions
- Fix issue with conditional sections being ended early

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
- Improve repeated sections
  - Add partial support for nested sections
    - Does not properly support forms with duplicate question IDs
  - Add support for repeated sections based on multiple questions

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
- Remove parent section from repeated section

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
- Add supportt for:
  - Address questions & onlyCountries, regions columns
  - Phone questions & countries, searchPlacesAround columns
  - Selectable Area questions & variant column
  - enableNotes column (True or False)
  - validationRegexp

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
- Fix error with labelless sections

sashaandjic commented:
- remove `'` and `][` from subject types

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
Add columns for info box type and validation error messages

marta- commented:
When generating a condition with unary comparator, do not generate `operandB`.

marta- commented:
For repeated sections, we need the ability to exclude some options when we generate the related section. For our immediate use cases, we can hardcode for now the exclusion of "None" and "None of the above" options.

sashaandjic commented:
unable to use `--max-per-subject` param
![image](https://github.com/data-team-uhn/cards/assets/23243244/c7bc993b-bbaa-4227-af91-596544a9ecd5)

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
- Add support for conditionals within repeated section:
  - If a conditional is in a repeated section, it will have all it's references renamed
    to match the edited naming scheme of the repeated section
- Fixed conditionals so single argument operators will not have operandB generated
- Fixed pagination arguments being multivalued
- Fixed subject types having extraneous brackets and quotes

acrowthe committed:
[MISC] Fix form xml import for Sparc forms
- Fix child of sections in repeated sections not getting renamed correctly

sashaandjic reviewed (approved):
tested

marta- commented:
This PR contains a lot of improvements already. I'm ok to merge it in its current state to facilitate project splitting, and to make a new PR if/when other changes are needed.

sashaandjic reviewed (approved):
tested with the newest `heracles` reports and most `sparc` reports
